example/example/core/views.py

```diff
@@ -17,7 +28,46 @@ def middleware_tester_table(request):
     return render(
         request,
         "middleware-tester-table.html",
+        {"timestamp": time.time()},
+    )
+
+
+# Partial rendering example
+
+
+# This dataclass acts as a stand-in for a database model - the example app
+# avoids having a database for simplicity.
+
+
+@dataclass
+class Person:
+    id: int
+    name: str
+
+
+faker = Faker()
+people = [Person(id=i, name=faker.name()) for i in range(1, 235)]
+
+
+@require_http_methods(("GET",))
+def partial_rendering(request):
+    # Standard Django pagination
+    page_num = request.GET.get("page", "1")
+    page = Paginator(object_list=people, per_page=10).get_page(page_num)
+
+    # The htmx magic - use a different, minimal base template for htmx
+    # requests, allowing us to skip rendering the unchanging parts of the
+    # template.
+    if request.htmx:
+        base_template = "_partial.html"
+    else:
+        base_template = "_base.html"
```

Comment:
This, combined with the `{% extends base_template %}` in the template file, removes the need for `{% htmx_include %}`

We could include a mixin for those using class-based views, but it would be very small (only adding one context variable) and maybe people would want to use a different variable name than `base_template`.